includes/class-donations.php

```diff
@@ -275,4 +304,59 @@ protected static function update_donation_product( $args ) {
 			$child_product->save();
 		}
 	}
+
+	/**
+	 * Handle submission of the donation form.
+	 */
+	public static function process_donation_form() {
+		$donation_form_submitted = filter_input( INPUT_GET, 'newspack_donate', FILTER_SANITIZE_NUMBER_INT );
```

Comment:
Doesn't this create a possibility that an unauthorized form submission could be interpreted as coming from the Donate block? One the one hand, this wouldn't be a significant problem since all we're doing is populating the cart, but it still could be seen as a problem. Could we solve  this with a nonce strategy?

Reply:
We could add a nonce, but it wouldn't really help much. The main reason I didn't worry about it is because [WooCommerce doesn't put a nonce on its generic `add-to-cart` handler](https://github.com/woocommerce/woocommerce/blob/master/includes/class-wc-form-handler.php#L726). e.g. anyone can put a link to `example.com?add-to-cart={product_id}` anywhere. There isn't any real security risk (other than slightly inconveniencing someone), and it lets people easily use the links in articles, emails, etc. (e.g. 'Click here to donate $10 now!')

Reply:
OK sounds good.